rust/lance/src/index/vector/ivf/io.rs

```diff
@@ -75,20 +86,31 @@ pub(super) async fn write_index_partitions(
 
     for part_id in 0..ivf.num_partitions() as u32 {
         let start = Instant::now();
-        let mut pq_array = Vec::<Arc<dyn Array>>::new();
-        let mut row_id_array = Vec::<Arc<dyn Array>>::new();
-
-        if let Some(existing_idx) = existing_partitions.as_ref() {
-            let part = existing_idx.load_partition(part_id as usize, true).await?;
-            let pq_idx = part.as_any().downcast_ref::<PQIndex>().unwrap();
-            if pq_idx.code.is_some() {
-                let pq_code_arr = pq_idx.code.as_ref().unwrap().clone();
-                let pq_code_fixed_size_arr = FixedSizeListArray::try_new_from_values(
-                    pq_code_arr.as_ref().clone(),
-                    pq_idx.pq.num_sub_vectors() as i32,
-                )?;
-                pq_array.push(Arc::new(pq_code_fixed_size_arr));
-                row_id_array.push(pq_idx.row_ids.as_ref().unwrap().clone());
+        let mut pq_array: Vec<Arc<dyn Array>> = vec![];
```

Comment:
nit: should we reserve capacity here?

Reply:
This is # of total sub index? it is less likely to be more than 100 ?